src/openforms/logging/logevent.py

```diff
@@ -437,6 +440,58 @@ def submission_export_list(form: "Form", user: "User"):
         user=user,
     )
 
+def submission_logic_evaluated(submission: "Submission", evaluated_logic, trigger = False):
+
+    formsteps = submission.form.formstep_set.all()
+
+    candidates_keys = []
+    for formstep in formsteps :
+        candidates_keys = candidates_keys + formstep.form_definition.get_all_keys()
+    
+    logic_test = JsonLogicTest.from_expression(evaluated_logic.json_logic_trigger)
+    source_components = formating_json_logic(logic_test, candidates_keys, submission)
+    
+    # Case only one leaf (i.e not advanced logic)
+    if isinstance(source_components, dict) :
+        source_components = [source_components]
+
+    targeted_components = list()
+    for action in evaluated_logic.actions :
+        log_action = {}
+
+        # Action verifications 
+        action_details = action["action"]
+        log_action['value'] = ""
+        log_action['type'] = action_details['type']
+        if action_details["type"] == LogicActionTypes.value:
+            log_action['value'] = action_details['value']
+            log_action['key'] = action['component']
+        elif action_details["type"] == LogicActionTypes.property:
+            log_action['type'] = action_details['type']
+            log_action['value'] = action_details['property']['value']
+            log_action['state'] = action_details['state']
+            log_action['key'] = action['component']
+        targeted_components.append(log_action)
+
+    # Assuming key is unique in all form steps 
+    for targeted_component in targeted_components :
+        # Adding the label of the key and step name associated if key exists 
+        if 'key' in targeted_component.keys() :
+            for formstep in formsteps :
+                if targeted_component['key'] in formstep.form_definition.get_all_keys() :
+                    targeted_component['label'] = formstep.form_definition.get_key_label(targeted_component['key'])
+                    targeted_component['step_name'] = formstep.form_definition.name
+                    break
```

Comment:
This can benefit from some guard clauses to remove indentation from your code:

```suggestion
    for targeted_component in targeted_components :
        # Adding the label of the key and step name associated if key exists 
        if not (key := target_component.get("key")):
            continue
        for formstep in formsteps:
            if key not in formstep.form_definition.get_all_keys():
                continue
            targeted_component['label'] = formstep.form_definition.get_key_label(key)
            targeted_component['step_name'] = formstep.form_definition.name
            break
```

This is an example of flattening the loops and using guard clauses - additionally I think it's better to build a mapping _before_ entering the actions loop where you map formio component key to the form step it belongs, because this information is a loop invariant and doesn't need to be re-calculated every time you introspect/enhance an action.